### PR TITLE
[NCL-5216] BuildTest integration tests

### PR DESCRIPTION
**Pull request by michalovjan**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

**Comment by michalovjan** [comment]
PR-tests were successful on the first try.... WAAAAAAAAAT

**Comment by thescouser89** [comment]
** Mind games **

File: integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/BuildEndpointTest.java [hunk 1]
```diff
@@ -239,7 +239,7 @@ public void shouldGetBuildLogs() throws ClientException {
 
     @Test
     public void shouldFailToGetSshCredentialsForUserThatDidntTrigger() {
-        BuildClient client = new BuildClient(RestClientConfiguration.asUser());
+        BuildClient client = new BuildClient(RestClientConfiguration.asSystem());
```
Comment (janinko) [inline integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/BuildEndpointTest.java]: Why did you cahnge this?

File: rest-api/src/main/java/org/jboss/pnc/rest/api/parameters/BuildParameters.java [hunk 2]
```diff
@@ -36,19 +36,15 @@
...
     @Parameter(description = "What should varant rebuild?")
     @QueryParam("rebuildMode")
+    @DefaultValue("IMPLICIT_DEPENDENCY_CHECK")
     RebuildMode rebuildMode;
 
     @Parameter(description = "Should we build also dependencies of this BuildConfiguration?")
     @QueryParam("buildDependencies")
     @DefaultValue("true")
-    boolean buildDependencies;
+    Boolean buildDependencies;
```
Comment (janinko) [inline rest-api/src/main/java/org/jboss/pnc/rest/api/parameters/BuildParameters.java]: what if it would be 
```suggestion
    boolean buildDependencies = true;
```
Would that work? Would that break something?

File: integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/BuildTest.java [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+
+    private static final Logger logger = LoggerFactory.getLogger(BuildTest.class);
+
+    BuildConfigurationClient buildConfigurationClient = new BuildConfigurationClient(asUser());
+
+    GroupConfigurationClient groupConfigurationClient = new GroupConfigurationClient(asUser());
+
+    GroupBuildClient groupBuildClient = new GroupBuildClient(asUser());
+
+    BuildClient buildClient = new BuildClient(asUser());
```
Comment (janinko) [inline integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/BuildTest.java]: private?

File: integration-test/src/test/java/org/jboss/pnc/integration_new/setup/RestClientConfiguration.java [hunk 4]
```diff
@@ -34,7 +34,7 @@ public static Configuration getConfiguration(AuthenticateAs authAs) {
         if (AuthenticateAs.SYSTEM_USER.equals(authAs)) {
             builder.basicAuth(new Configuration.BasicAuth("system", "system.1234"));
         } else if (AuthenticateAs.USER.equals(authAs)) {
-            builder.basicAuth(new Configuration.BasicAuth("user", "pass.1234"));
+            builder.basicAuth(new Configuration.BasicAuth("demo-user", "pass.1234"));
```
Comment (janinko) [inline integration-test/src/test/java/org/jboss/pnc/integration_new/setup/RestClientConfiguration.java]: Why did you change this?

File: rest-api/src/main/java/org/jboss/pnc/rest/api/parameters/BuildParameters.java (hunk 2 above)
Comment (michalovjan) [reply rest-api/src/main/java/org/jboss/pnc/rest/api/parameters/BuildParameters.java]: Then you would always have buildDependencies=true in a request. I don't have a strong opinion on this.

File: integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/BuildEndpointTest.java (hunk 1 above)
Comment (michalovjan) [reply integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/BuildEndpointTest.java]: I had to change authentication to "demo-user" because as old "user" when building inside the code I would have User as null because we don't have user with username:"user" in our demo-data. 

In this case that buildRecord, according to demo-data, was built by demo-user so I had to change the user in request. The "system" was our only other user to choose from.

File: integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/BuildTest.java (hunk 3 above)
Comment (michalovjan) [reply integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/BuildTest.java]: ya, mb

File: integration-test/src/test/java/org/jboss/pnc/integration_new/setup/RestClientConfiguration.java (hunk 4 above)
Comment (michalovjan) [reply integration-test/src/test/java/org/jboss/pnc/integration_new/setup/RestClientConfiguration.java]: I had to change authentication to "demo-user" because as old "user" when building inside the code I would have User as null because we don't have user with username:"user" in our demo-data.
